llvm/lib/ProfileData/InstrProf.cpp

```diff
@@ -640,9 +641,11 @@ Error InstrProfSymtab::addFuncWithName(Function &F, StringRef PGOFuncName) {
   if (Error E = NameToGUIDMap(PGOFuncName))
     return E;
 
-  StringRef CanonicalFuncName = getCanonicalName(PGOFuncName);
-  if (CanonicalFuncName != PGOFuncName)
-    return NameToGUIDMap(CanonicalFuncName);
+  if (AddCanonical) {
```

Comment:
Maybe early return with an inverted condition to avoid nesting below?

`if(!AddCanonical) return Error::success`

Reply:
done